src/controller/AutoCommissioner.cpp

```diff
@@ -510,43 +510,33 @@ CHIP_ERROR AutoCommissioner::CommissioningStepFinished(CHIP_ERROR err, Commissio
         case CommissioningStage::kSendDACCertificateRequest:
             SetDAC(report.Get<RequestedCertificate>().certificate);
             break;
-        case CommissioningStage::kSendAttestationRequest:
-            if (report.Get<AttestationResponse>().attestationElements.size() > kAttestationElementsLength)
+        case CommissioningStage::kSendAttestationRequest: {
+            auto & elements  = report.Get<AttestationResponse>().attestationElements;
+            auto & signature = report.Get<AttestationResponse>().signature;
+            if (elements.size() > sizeof(mAttestationElements))
```

Comment:
You cannot compare against a sizeof of a native struct. AttestationElements is TLV.

All of this debug code seems odd. If there is a bug let's fix it. Let's not litter "just in case the APIs don't fulfill the API contract requirements" debug.

How could this happen? Did you observe it happen?

Having a garbage span used-after-free is not going to be fully resolved by this. If the garbage is small enough, then it will get through this.

Reply:
we are copying the raw TLV into mAttestationElements so why doesn't the size check make sense?